private-templates-service/adaptivecards-templating-service/src/storageproviders/MongoDBProvider.ts

```diff
@@ -0,0 +1,198 @@
+import { StorageProvider } from "./IStorageProvider";
+import { IUser, ITemplate, JSONResponse, ITemplateInstance } from "../models/models";
+import { MongoUtils } from "../util/mongoutils/mongoutils";
+import { MongoWorker } from "../util/mongoutils/MongoWorker";
+
+const defaultConnectionOptions = {
+  useNewUrlParser: true,
+  useCreateIndex: true,
+  useUnifiedTopology: true,
+  connectTimeoutMS: 5000,
+  socketTimeoutMS: 30000,
+  useFindAndModify: false
+};
+export class MongoDBProvider implements StorageProvider {
+  worker: MongoWorker;
+
+  constructor(connectionString: string, options: any = defaultConnectionOptions) {
+    this.worker = new MongoWorker(connectionString, options);
+  }
+
+  // Construct functions are introduced to be able to search by
+  // nested objects. For example, if team:["merlin", "morgana"] are
+  // provided in the query, then we need to add operator $all
+  // to make sure they're both in the returned user 'team' field
+  private _constructUserQuery(query: Partial<IUser>): any {
+    let userQuery: any = { ...query };
+    if (query.team) {
+      userQuery.team = { $all: query.team };
+    }
+    if (query.org) {
+      userQuery.org = { $all: query.org };
+    }
+    return userQuery;
+  }
+  private _constructTemplateQuery(query: Partial<ITemplate>): any {
+    let templateQuery: any = { ...query };
+    if (query.tags) {
+      templateQuery.tags = { $all: query.tags };
+    }
+    return templateQuery;
+  }
+  async getUsers(query: Partial<IUser>): Promise<JSONResponse<IUser[]>> {
```

Comment:
When no users exist, should getUsers return false or true? I think the MongoDBProvider returns false, but the InMemoryDBProvider returns true - might be useful to add comments to the StorageProvider so that all providers return consistent results and it's easier for the backend to use

Reply:
Yes, it needs to be fixed. I'll add the comments to the IStorageProvider to indicate expected behaviour.